logbook-api/src/main/java/org/zalando/logbook/RequestURI.java

```diff
@@ -63,9 +64,12 @@ private static String reconstruct(final BaseHttpRequest request, final Set<Compo
         if (components.contains(AUTHORITY)) {
             url.append("//").append(host);
 
-            if (isNotStandardPort(scheme, port)) {
-                url.append(':').append(port);
-            }
+            port.ifPresent(p -> {
+                if (isNotStandardPort(scheme, p)) {
```

Comment:
> What if I run http on 443 or https on 80?

This should be handled correctly.

> Should we care here at all? Maybe just append if present

I agree, now that we know whether a port was specified, we don't need the reconstruction logic.